src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/If_AllowsSideEffects.txt

```diff
@@ -0,0 +1,20 @@
+﻿#SETUP: EnableExpressionChaining
+
+
+// These test cases are part of the code coverage and are meant to reach IfFunction.IsArgTypeInconsequential
+
+// This If() expression is allowed to have mismatching types between arg1 and arg2 because
+// a) it's behavior,
+// b) it's verified nobody is actually using the return value.
+
+>> If(false, "abc1", {x:3})
+{x:3}
+
+>> If(false, "abc2", {x:3}; {x:2}); Value(100)
+100
+
+>> If(false, "abc3", {x:3}; If(true, Value(10); {x:3})); Value(101)
+101
+
+>> Value(If(false, "abc4", {x:3}; If(true, Value(10); {x:3})); Value(102))
+102
```

Comment:
We should add tests where the If value is used (e.g., `If(1>0,100,{x:3}) + 1` - in this case we should have a compile-time error